crypto-core/src/test/java/com/palantir/crypto2/io/DecryptingSeekableInputTest.java

```diff
@@ -62,18 +64,37 @@ public static void beforeClass() throws IOException {
     }
 
     @Parameterized.Parameters
-    public static Collection<String> ciphers() {
-        return ImmutableList.of(AesCtrCipher.ALGORITHM, AesCbcCipher.ALGORITHM);
+    public static Collection<Pair<String, BiFunction<SeekableCipher, SeekableInput, SeekableInput>>> ciphers() {
+        return ImmutableList.of(
+                new Pair<>(AesCtrCipher.ALGORITHM, DecryptingSeekableInputTest::apacheStream),
+                new Pair<>(AesCtrCipher.ALGORITHM, DecryptingSeekableInputTest::jceStream),
+                new Pair<>(AesCbcCipher.ALGORITHM, DecryptingSeekableInputTest::jceStream));
     }
 
-    public DecryptingSeekableInputTest(String cipher) {
+    private static SeekableInput apacheStream(SeekableCipher cipher, SeekableInput input) {
+        if (cipher instanceof AesCtrCipher) {
+            return ApacheCtrDecryptingSeekableInput.create(input, cipher.getKeyMaterial());
```

Comment:
modified so this class can only be constructed with OpenSSL and won't fall back to JCE. also added an `isSupported` method that checks if OpenSSL was loaded.